agent/command/git.go

```diff
@@ -137,19 +137,34 @@ func (opts *cloneOpts) setLocation() error {
 
 // getProjectMethodAndToken returns the project's clone method and token. If
 // set, the project token takes precedence over GitHub App token which takes precedence over over global settings.
-func getProjectMethodAndToken(projectToken, globalToken, appToken, globalCloneMethod string) (string, string, error) {
+func getProjectMethodAndToken(ctx context.Context, comm client.Communicator, td client.TaskData, owner, repo, projectToken, globalToken, globalCloneMethod string) (string, string, error) {
 	if projectToken != "" {
 		token, err := parseToken(projectToken)
 		return evergreen.CloneMethodOAuth, token, err
 	}
-	if appToken != "" {
-		token, err := parseToken(appToken)
-		return evergreen.CloneMethodAccessToken, token, err
+
+	appToken, err := comm.CreateInstallationToken(ctx, td, owner, repo)
+	// TODO EVG-21022: Remove fallback once we delete GitHub tokens as expansions.
+	if err != nil {
+		grip.Warning(message.WrapError(err, message.Fields{
+			"message": "error creating GitHub app token, falling back to legacy clone methods",
+			"owner":   owner,
+			"repo":    repo,
+			"task":    td.ID,
+			"ticket":  "EVG-21022",
+		}))
+	} else if appToken == "" {
```

Comment:
nit: might read more easily if-elseif was simplified down so the debug didn't need an explicit case:
```go
appToken, err := comm.CreateInstallationToken(...)
grip.Warning(...)
if appToken != "" {
    return evergreen.CloneMethodAccessToken, appToken, nil
}

grip.Debug(...)

// Continue with fallback
```